dm/worker/subtask.go

```diff
@@ -73,9 +73,13 @@ type SubTask struct {
 	l log.Logger
 
 	sync.RWMutex
-	wg     sync.WaitGroup
+	wg sync.WaitGroup
+	// ctx is used for the whole subtask. It will be created only when we new a subtask.
 	ctx    context.Context
 	cancel context.CancelFunc
+	// currCtx is used for one loop. It will be created each time we use st.run/st.Resume
+	currCtx    context.Context
```

Comment:
Is it enough to just let `ctx` as `used for one loop`

Reply:
I don't think it's enough. Suggest a situation:
1. we close a subtask and remove subtask from subtask holder.
2. After this subtask's cancel was runned, a goroutine build a new ctx and cancel and start `subtask.Run`.
If this worker is not stopped, this subtask will keep running without monitor. It's dangerous.